src/main/java/com/kth/ciserver/controller/CiController.java

```diff
@@ -27,13 +46,48 @@ public String handleGithubWebhook(@RequestBody GithubWebhookRequest request) thr
 
     private void pullAndBuildApplication(GithubWebhookRequest request) throws IOException {
         executeAndPrintCommand("git pull");
-        executeAndPrintCommand(String.format("git checkout %s", request.getHead_commit().getId()));
+        executeAndPrintCommand(String.format("git checkout %s", request.getHeadCommit().getId()));
         Boolean testsSuccessful = executeAndPrintCommand("./gradlew clean build");
         executeAndPrintCommand("git checkout main");
         System.out.println("Tests successful: " + testsSuccessful); // this should instead be printed to file for consistent storage
+
+        updateGithubCommitStatus(testsSuccessful, request.getHeadCommit().getId(), request.getRepository().getStatusesUrl());
+    }
+
+    private void updateGithubCommitStatus(Boolean testsSuccessful, String commitId, String statusesUrl) {
+        String commitState;
+
+        if (testsSuccessful)
+            commitState = COMMIT_STATE_SUCCESS;
+        else
+            commitState = COMMIT_STATE_FAILURE;
+
+        String url = statusesUrl.replace("{sha}", commitId);
+
+        HttpStatus httpStatus = postRequest(url, commitState);
+        System.out.println("Http status for post to Github " + httpStatus);
+    }
+
+    private HttpStatus postRequest(String url, String commitState) {
```

Comment:
I really want to seperate this out into a different file, but since we are all working at the same time I will wait until the end to refactor